web3/_utils/module.py

```diff
@@ -18,11 +18,13 @@
 
 def attach_modules(
     parent_module: Union["Web3", "Module"],
-    module_definitions: Dict[str, Sequence[Any]],
+    module_definitions: Dict[str, Any],
     w3: Optional[Union["Web3", "Module"]] = None
 ) -> None:
     for module_name, module_info in module_definitions.items():
-        module_class = module_info[0]
+        module_info_is_list_like = isinstance(module_info, Sequence)
```

Comment:
I think we still need to allow the tuple form: `{'eth': (Eth,)}` so that we're not introducing a breaking change here, which definitely makes this method uglier. But we can remove the top-level tuple allowance in v6; we just need to add a comment to the v6 breaking changes issue or create an issue and link it in the v6 breaking changes issue.

Reply:
For sure, did not want to introduce a breaking change. The tuple can still be passed in. This just makes it so both can be passed in. And since it can be simplified, I just changed it in the docs.

`{'eth': (Eth,)}` works but just seems like it is more complicated than needs to be if the module doesn't have submodules. So these changes also allow `{'eth': Eth}` if it is a simple module.

Reply:
Oh sweet. Do you mind adding a test then to make sure we don't regress there?

Reply:
> Oh sweet. Do you mind adding a test then to make sure we don't regress there?
> 
> Edit: Or maybe there is a test and I just missed it

Good point... I think that can be useful. Though in the tests we do pass in tuples for the modules that do have submodules so in theory it should work with a tuple with a single class. But this isn't very obvious so I can add a simple test.